FUNPY-26 Improves snapshots sync time

nessv opened the pull request:
<!-- Fill in the following sections, deleting any sections that you leave blank -->

# Description <!-- [OPTIONAL] -->
Improves the sync time for snapshots

# Changes <!-- [REQIURED] -->
<!-- Fill in a bulleted list with changes made in this pull request -->
- New GET call to retrieve all the snapshots asociated with a family.
- Replaces the GET request that retrieves all the families, this in order to fix a
bug that was causing to show a lot more families compare with the web application.

# Tests <!-- [REQUIRED] -->
<!-- Check off the following tests (using [X]) that you performed to ensure that your changes work -->
This code has been tested in the following ways:
- Tasks:
  - [X] Logged in to the application
  - [X] Fill out a snapshot for a new family
  - [X] Fill out a snapshot for an existing family
  - [ ] Fill out priorities for a snapshot
  - [X] View a family and it's priorities
- Screen Size:
  - [ ] 7" screen size
  - [X] 8" screen size
  - [ ] 10" screen size

nessv committed:
FUNPY-26 Improves snapshots sync time

- New GET call to retrieve all the snapshots asociated with a family.
- Replaces the GET request that retrieves all the families, this in order to fix
a bug that was causing to retrieve a lot more families compare with the web application.

alefq commented on app/src/main/java/org/fundacionparaguaya/adviserplatform/data/remote/FamilyService.java:
```diff
@@ -17,8 +17,8 @@
-    @GET("families")
-    Call<List<FamilyIr>> getFamilies();
+    @GET("families/user")
+    Call<List<FamilyIr>> getFamilies(@Query("name") String name);
```
This should keep uniformity with API Path and FP-PSP-SERVER names. 
Rename the method to `getFamiliesByUserAndNameOrCode`, and `name` parameter to `nameOrCode`.

nessv replied:
I will change the method name as well as the variable name. But on the server side is expecting `name`, so I will leave `@Query("name")` as it is.

alefq commented on app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java:
```diff
@@ -278,32 +278,23 @@ private boolean pullSnapshots() {
-                Log.d(TAG, stopWatch.lap(String.format("Family: %s, Survey: %s", family.getName(),
-                        survey.getDescription())));
-                success &= pullSnapshots(family, survey);
-                if(success && getDashActivity() != null) {
```
Restore this `DashActivity` progress indicator maintaining coherence between `recordsCount` from `BaseRepository` and local `loopCount`

nessv replied:
I will restore `DashActivity` as requested 👌🏻

alefq commented on app/src/main/java/org/fundacionparaguaya/adviserplatform/data/repositories/SnapshotRepository.java:
```diff
@@ -346,7 +337,7 @@ private boolean pullSnapshots(Family family, Survey survey)
                     }
                     snapshot.setPriorities(
-                            IrMapper.mapPriorities(prioritiesResponse.body(), survey));
+                            IrMapper.mapPriorities(prioritiesResponse.body(), surveyList.get(snapshot.getSurveyId() - 1)));
```
This causes errors on Test Server

```
06-22 14:30:51.045 10626-10960/org.fundacionparaguaya.adviserplatform E/SyncManager: sync: Error while syncing!
    java.lang.IndexOutOfBoundsException: Index: 19, Size: 5
        at java.util.ArrayList.get(ArrayList.java:437)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.pullSnapshots(SnapshotRepository.java:341)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.pullSnapshots(SnapshotRepository.java:285)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SnapshotRepository.sync(SnapshotRepository.java:371)
        at org.fundacionparaguaya.adviserplatform.data.repositories.BaseRepository.sync(BaseRepository.java:39)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SyncManager.resyncWithOneAuthentication(SyncManager.java:189)
        at org.fundacionparaguaya.adviserplatform.data.repositories.SyncManager.sync(SyncManager.java:136)
        at org.fundacionparaguaya.adviserplatform.jobs.SyncJob.onRunJob(SyncJob.java:62)
        at com.evernote.android.job.Job.runJob(Job.java:132)
        at com.evernote.android.job.JobExecutor$JobCallable.runJob(JobExecutor.java:191)
        at com.evernote.android.job.JobExecutor$JobCallable.call(JobExecutor.java:176)
        at com.evernote.android.job.JobExecutor$JobCallable.call(JobExecutor.java:159)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
```

nessv replied:
I'm not been able to reproduce it. Under what circumstances did you get it?

nessv committed:
Restores DashActivity and renames method

This commit contains the follow:
- Restores DashActivity to display the current number of family snapshots
that are been syncronize.
- Changes the GET call `getFamilies` to `getFamiliesByUserAndNameOrCode`
as well as the variable `name` to `nameOrCode`

nessv commented:
@alefq I made a new commit with the changes you requiere. I wasn't able to reproduce the error you pointed up (I've tested locally as well as in Test enviorment). Please can you confirm that the error persists?

nessv committed:
Implements new approach to get current the survey

Simply iterates over the survey_list
